addons/product/models/product_attribute.py

```diff
@@ -128,19 +186,48 @@ class ProductTemplateAttributeLine(models.Model):
     _description = 'Product Template Attribute Line'
     _order = 'attribute_id, id'
 
-    product_tmpl_id = fields.Many2one('product.template', string='Product Template', ondelete='cascade', required=True, index=True)
-    attribute_id = fields.Many2one('product.attribute', string='Attribute', ondelete='restrict', required=True, index=True)
-    value_ids = fields.Many2many('product.attribute.value', string='Attribute Values')
+    product_tmpl_id = fields.Many2one('product.template', string="Product Template", ondelete='cascade', required=True, index=True,
+        states={'created': [('readonly', True)]})
+    attribute_id = fields.Many2one('product.attribute', string="Attribute", ondelete='restrict', required=True, index=True,
+        states={'created': [('readonly', True)]})
+    value_ids = fields.Many2many('product.attribute.value', string="Values", domain="[('attribute_id', '=', attribute_id)]",
+        relation='product_attribute_value_product_template_attribute_line_rel')
     product_template_value_ids = fields.Many2many(
         'product.template.attribute.value',
         string='Product Attribute Values',
         compute="_set_product_template_value_ids",
         store=False)
+    state = fields.Selection(compute='_compute_state',
```

Comment:
do not create a field to just check if the record exist or not, use `record.exists()`.